pkgs/tools/package-management/nix/common-meson.nix

```
@@ -0,0 +1,286 @@
+{
+  lib,
+  fetchFromGitHub,
+  version,
+  suffix ? "",
+  hash ? null,
+  src ? fetchFromGitHub {
+    owner = "NixOS";
+    repo = "nix";
+    rev = version;
+    inherit hash;
+  },
+  patches ? [ ],
+  maintainers ? lib.teams.nix.members ++ [
+    lib.maintainers.lovesegfault
+    lib.maintainers.artturin
+  ],
+  self_attribute_name,
+}@args:
+assert (hash == null) -> (src != null);
+{
+  stdenv,
+  bison,
+  boehmgc,
+  boost,
+  brotli,
+  busybox-sandbox-shell,
+  bzip2,
+  cmake,
+  curl,
+  darwin,
+  doxygen,
+  editline,
+  flex,
+  git,
+  gtest,
+  jq,
+  lib,
+  libarchive,
+  libcpuid,
+  libgit2,
+  libsodium,
+  lowdown,
+  lowdown-unsandboxed,
+  toml11,
+  man,
+  meson,
+  ninja,
+  mdbook,
+  mdbook-linkcheck,
+  nlohmann_json,
+  nixosTests,
+  openssl,
+  perl,
+  perlPackages,
+  python3,
+  pkg-config,
+  rapidcheck,
+  rsync,
+  Security,
+  sqlite,
+  util-linuxMinimal,
+  xz,
+  enableDocumentation ? stdenv.buildPlatform.canExecute stdenv.hostPlatform,
+  enableStatic ? stdenv.hostPlatform.isStatic,
+  withAWS ? !enableStatic && (stdenv.hostPlatform.isLinux || stdenv.hostPlatform.isDarwin),
+  aws-sdk-cpp,
+  withLibseccomp ? lib.meta.availableOn stdenv.hostPlatform libseccomp,
+  libseccomp,
+
+  confDir,
+  stateDir,
+  storeDir,
+
+  # passthru tests
+  pkgsi686Linux,
+  pkgsStatic,
+  runCommand,
+  pkgs,
+}:
+let
+  self = stdenv.mkDerivation {
```

I believe it's better to use `finalAttrs.finalPackage` rather than a let-bound `self`, isn't it?
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

---

Why would that not work with `finalAttrs.finalPackage`?

---

I didn't know about finalPackage.